### PR TITLE
chore: Initial setup of lerna

### PR DESCRIPTION
**Pull request by pksunkara**
Haven't tested publishing yet, will add it to readme after that is done and finalized.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,21 +0,0 @@
-# Contributing Guide
```
Comment (kylef) [inline CONTRIBUTING.md]: Why was this removed? We can keep these instructions for contributing to the docs and extend it in future for other parts?

File: .travis.yml [hunk 2]
```diff
@@ -1,4 +0,0 @@
-language: node_js
```
Comment (kylef) [inline .travis.yml]: No CI?

Comment (pksunkara) [reply .travis.yml]: Will add it back later.

File: CONTRIBUTING.md (hunk 1 above)
Comment (pksunkara) [reply CONTRIBUTING.md]: Readme wont contain anything else. So, I moved all of contributing.md to readme

**Comment by pksunkara** [comment]
Addressed.

**Comment by honzajavorek** [comment]
You are using yarn?

**Comment by pksunkara** [comment]
I meant lerna has the option to use yarn workspaces feature which is way faster than npm. The difference was night and day since I had the same thought as you and was using npm initially when I was experimenting with lerna.

File: README.md [hunk 3]
```diff
@@ -1,32 +1,50 @@
-# API Elements JavaScript
+# API Elements
 
-This library provides an interface to [API Elements](http://apielements.org) in JavaScript.
```
Comment (honzajavorek) [inline README.md]: Why this line was deleted? I think the README should have an introductory ~paragraph~ sentence.

File: README.md [hunk 4]
```diff
@@ -1,32 +1,50 @@
...
+Install dependencies of all packages:
 
 ```shell
-$ npm install api-elements
+$ npm install --global yarn
+$ yarn
 ```
 
-## Usage
+> **NOTE**: Please do commit the yarn.lock to the github repo
```
Comment (honzajavorek) [inline README.md]:
```suggestion
> **NOTE**: Please do commit the `yarn.lock` to the GitHub repo
```

File: README.md [hunk 5]
```diff
@@ -1,32 +1,50 @@
...
-// Parsing a JSON Representation of API Elements tree
-const parseResult = namespace.serialiser.deserialise({
-  element: 'parseResult',
-  content: []
-});
+## Documentation
 
-console.log(parseResult);
+The documentation is built using Sphinx, a Python tool. Assuming you have
+Python 3 installed, the following steps can be used to build the site.
```
Comment (honzajavorek) [inline README.md]: > can be used to build the site

The sentence isn't entirely true. The following steps get you as far as having the docs stack installed, but doesn't give you any clues on the further workflow.

File: README.md [hunk 6]
```diff
@@ -1,32 +1,50 @@
...
 
-// Creating API Elements directly
-const parseResult = new namespace.elements.ParseResult();
-console.log(parseResult);
+### Running the Development Server
+
+You can run a local development server to preview changes using the following:
+
+```shell
+$ make watch
```
Comment (honzajavorek) [inline README.md]: Wouldn't it be nicer to have npm scripts for this? Also, I haven't checked the `requirements.txt`, but if you use the same dependencies as in https://github.com/apiaryio/dredd/pull/1119, they're all approved and then you can basically copy-paste the Dredd setup, if you like it.

https://github.com/apiaryio/dredd/blob/c2dadfbd9921725f817f23783a3ee15ca6753546/package.json#L18:L19

Up to you if this is something for a follow-up PR, I don't want to block on this one.

File: package.json [hunk 7]
```diff
@@ -1,26 +1,17 @@
...
+    "test": "lerna exec --scope='*' -- npm run test",
+    "lint": "lerna exec --scope='*' -- npm run lint"
   },
   "devDependencies": {
-    "chai": "^4.1.2",
-    "mocha": "^5.0.2"
-  },
-  "engines": {
-    "node": ">=4"
+    "lerna": "^3.4.0"
```
Comment (honzajavorek) [inline package.json]: Do we want to use `^`?

File: README.md (hunk 3 above)
Comment (honzajavorek) [reply README.md]: I understand now, the original README has been moved to the sub-package. Well still the root README should have an introductory sentence on what it contains.

File: packages/api-elements/package.json [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+    "url": "https://github.com/apiaryio/api-elements.js.git"
+  },
+  "scripts": {
+    "test": "mocha"
+  },
+  "author": "Apiary.io <support@apiary.io>",
+  "license": "MIT",
+  "dependencies": {
+    "minim": "^0.20.5",
+    "minim-parse-result": "^0.10.1"
```
Comment (honzajavorek) [inline packages/api-elements/package.json]: Do we want to use `^`-s? (comment applies to all deps in this file)

**Comment by honzajavorek** [comment]
Also, if Yarn isn't required by users of API Elements and it brings benefit (not overhead) to the maintainers to this package, then I approve. OTOH I'd prefer to have this decision (the why) explicitly documented in the README (or any kind of contributing / maintainers' docs).

File: package.json (hunk 7 above)
Comment (pksunkara) [reply package.json]: Yes

**Comment by pksunkara** [comment]
@kylef Some of HJ's comments are about api-elements.js npm package

File: README.md (hunk 3 above)
Comment (kylef) [reply README.md]: Yes, I think that can come in future PRs this is part of larger task of moving Fury-* into the repository, we'll also have API Reference docs for all of them once it is done.

**Comment by kylef** [comment]
@honzajavorek Regarding the documentation, I've created https://github.com/apiaryio/api-elements.js/issues/17 with the changes you mentioned and some more.

File: README.md (hunk 3 above)
Comment (kylef) [reply README.md]: Moved to https://github.com/apiaryio/api-elements.js/issues/17

File: README.md (hunk 6 above)
Comment (kylef) [reply README.md]: It would, https://github.com/apiaryio/api-elements.js/issues/17 now mentions integrating the docs with NPM.

**Pull request merged**